var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -92,13 +102,22 @@ def cmake_args(self):
         spec = self.spec
 
         args = [
-            '-DADIOS2_BUILD_SHARED_LIBS:BOOL={0}'.format(
+            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
                 'ON' if '+shared' in spec else 'OFF'),
+            '-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL={0}'.format(
```

Comment:
This should be conditional on `~shared`.  Essentially you want to prevent setting shared w/o pic since that's an invalid configuration.